File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//function prototypes
int count_letters(string word);
int count_words(string text);
int count_sentences(string text);
void grade(float letters, float words, float sentences);

int main(void)
{
    string text;
    int letters, words, sentences;

    //prompt user for input
    do
    {
        text = get_string("Text: ");
    }
    while (strlen(text) == 0);

    //assign values to variables
    letters = count_letters(text);
    words = count_words(text);
    sentences = count_sentences(text);

    //print grade of text
    grade((float)letters, (float)words, (float)sentences);

}

//counting number of letters
int count_letters(string word)
{
    int letters = 0;

    for (int i = 0; i < strlen(word); i++)
    {
        if (isalpha(word[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

//counting number of words
int count_words(string text)
{
    int words = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            words ++;
        }
    }
    words++;
    return words;
}

//counting number of sentences
int count_sentences(string text)
{
    int sentence = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }
    return sentence;
}

//calculating grade of Text
void grade(float letters, float words, float sentences)
{
    double L, S;

    L = letters / words * 100;
    S = sentences / words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
# TODO

def main():
    text = ""

    # prompt user for input
    while (len(text) == 0):
        text = input("Text: ")

    # assign values to variables
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # print grade of text
    grade(letters, words, sentences)


# counting number of letters
def count_letters(word):
    letters = 0
    for i in range(len(word)):
        if (word[i] >= "a" and word[i] <= "z") or (word[i] >= "A" and word[i] <= "Z"):
            letters += 1
    return letters


# counting number of words
def count_words(text):
    words = 0
    for i in range(len(text)):
        if (text[i] == " "):
            words += 1
    words += 1
    return words


# counting number of sentences
def count_sentences(text):
    sentence = 0
    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            sentence += 1
    return sentence


# calculating grade of Text
def grade(letters, words, sentences):
    L = letters / words * 100
    S = sentences / words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if (index > 16):
        print("Grade 16+")
    elif (index < 1):
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()

